nixos/tests/pam/test_chfn.py

```diff
@@ -0,0 +1,24 @@
+expected_lines = {
+    "account required pam_unix.so",
+    "account sufficient /nix/store/wba7p4ks843gb8178vwi17vq11ngilfb-pam-krb5-4.10/lib/security/pam_krb5.so",
+    "auth [default=die success=done] /nix/store/gqa8nv64icrv7ds156cyf7l3l1ch263c-pam_ccreds-10/lib/security/pam_ccreds.so action=validate use_first_pass",
+    "auth [default=ignore success=1 service_err=reset] /nix/store/wba7p4ks843gb8178vwi17vq11ngilfb-pam-krb5-4.10/lib/security/pam_krb5.so use_first_pass",
+    "auth required pam_deny.so",
+    "auth sufficient /nix/store/gqa8nv64icrv7ds156cyf7l3l1ch263c-pam_ccreds-10/lib/security/pam_ccreds.so action=store use_first_pass",
+    "auth sufficient pam_rootok.so",
+    "auth sufficient pam_unix.so   likeauth try_first_pass",
+    "password sufficient /nix/store/wba7p4ks843gb8178vwi17vq11ngilfb-pam-krb5-4.10/lib/security/pam_krb5.so use_first_pass",
+    "password sufficient pam_unix.so nullok sha512",
+    "session optional /nix/store/wba7p4ks843gb8178vwi17vq11ngilfb-pam-krb5-4.10/lib/security/pam_krb5.so",
+    "session required pam_env.so conffile=/etc/pam/environment readenv=0",
+    "session required pam_unix.so",
+}
+actual_lines = set(machine.succeed("cat /etc/pam.d/chfn").splitlines())
```

Comment:
If you want to you could add more modules and also test more configuration files to get a good coverage

such as pam.d/login, pam.d/sudo, etc and some modules from https://search.nixos.org/options?channel=21.05&from=0&size=50&sort=relevance&type=packages&query=security.pam

Reply:
That's the plan, as soon as I get enough feedback on this that I'm fairly certain the result can be merged without major rework.